src/blocking.rs

```diff
@@ -0,0 +1,2830 @@
+//! OCI distribution client for fetching oci images from an OCI compliant remote store
+use std::collections::HashMap;
+use std::convert::TryFrom;
+use std::io::Write;
+use std::time::Duration;
+
+use http::header::RANGE;
+use http::HeaderValue;
+use http_auth::{parser::ChallengeParser, ChallengeRef};
+use olpc_cjson::CanonicalFormatter;
+use reqwest::blocking::{RequestBuilder, Response};
+use reqwest::header::HeaderMap;
+use reqwest::{NoProxy, Proxy, Url};
+use serde::Serialize;
+use tracing::{debug, trace, warn};
+
+pub use crate::types::*;
+
+use crate::digest::{digest_header_value, validate_digest, Digest, Digester};
+use crate::errors::*;
+use crate::manifest::{
+    ImageIndexEntry, OciImageIndex, OciImageManifest, OciManifest, Versioned,
+    IMAGE_MANIFEST_LIST_MEDIA_TYPE, IMAGE_MANIFEST_MEDIA_TYPE, OCI_IMAGE_INDEX_MEDIA_TYPE,
+    OCI_IMAGE_MEDIA_TYPE,
+};
+use crate::secrets::RegistryAuth;
+use crate::secrets::*;
+use crate::sha256_digest;
+use crate::token_cache::{RegistryOperation, RegistryToken, RegistryTokenType, SyncTokenCache};
+use crate::Reference;
+
+const MIME_TYPES_DISTRIBUTION_MANIFEST: &[&str] = &[
+    IMAGE_MANIFEST_MEDIA_TYPE,
+    IMAGE_MANIFEST_LIST_MEDIA_TYPE,
+    OCI_IMAGE_MEDIA_TYPE,
+    OCI_IMAGE_INDEX_MEDIA_TYPE,
+];
+
+const PUSH_CHUNK_MAX_SIZE: usize = 4096 * 1024;
+
+/// Default value for `ClientConfig::max_concurrent_upload`
+pub const DEFAULT_MAX_CONCURRENT_UPLOAD: usize = 16;
+
+/// Default value for `ClientConfig::max_concurrent_download`
+pub const DEFAULT_MAX_CONCURRENT_DOWNLOAD: usize = 16;
+
+/// Default value for `ClientConfig:default_token_expiration_secs`
+pub const DEFAULT_TOKEN_EXPIRATION_SECS: usize = 60;
+
+static DEFAULT_USER_AGENT: &str = concat!(env!("CARGO_PKG_NAME"), "/", env!("CARGO_PKG_VERSION"));
```

Comment:
Let's move this out of here and in `client.rs` to `lib.rs` so it isn't duplicated across both blocking and the normal client